api/reviews.js

```diff
@@ -1,4 +1,117 @@
 const reviewsRouter = require("express").Router();
-//insert query requests here//
+const {
+  getAllReviews,
+  getReviewById,
+  getReviewsByUser,
+  getReviewsByProduct,
+  createReview,
+  updateReview,
+  deleteReview,
+} = require("../db");
+const { requireUser } = require("./utils");
+
+//Gather all available reviews along with the products that they are in reference to.
+reviewsRouter.get("/", async (req, res, next) => {
+  try {
+    const reviews = await getAllReviews();
+    res.send(reviews);
+  } catch (error) {
+    console.error(error);
+    next({
+      name: "fetchReviewError",
+      message: "Cannot get all reviews",
+    });
+  }
+});
+
+//User should be able to pull all of their created reviews.
+reviewsRouter.get("/username/:username", async (req, res, next) => {
+  const { username } = req.params;
+  try {
+    const reviewByUser = await getReviewsByUser(username);
+    res.send(reviewByUser);
+  } catch (error) {
+    console.error(error);
+    next({
+      name: "noExistingReviews",
+      message: "There are no reviews under that username",
+    });
+  }
+});
+
+//Guest should be able to view all reviews of a particular product
+reviewsRouter.get("/product/:productId", async (req, res, next) => {
+  const { productId } = req.params;
+  try {
+    const reviewByProduct = await getReviewsByProduct(productId);
+    res.send(reviewByProduct);
+  } catch (error) {
+    console.error(error);
+    next({
+      name: "noExistingReviews",
+      message: "There are no reviews for the product",
+    });
+  }
+});
+
+//User must be able to create a review for a particular product 
+reviewsRouter.post("/", requireUser, async (req, res, next) => {
+  const { userId, productId, description, rating } = req.body;
+  try {
+    const newReview = await createReview({
+      userId: userId,
+      productId: productId,
```

Comment:
you don't have use the : format if the name and the value are the same
```suggestion
      userId,
      productId,
```